metasrv/src/meta_service/raftmeta.rs

```diff
@@ -265,33 +292,22 @@ impl MetaNode {
         let sto = Arc::new(sto);
 
         let mut builder = MetaNode::builder(&config).sto(sto.clone());
+        // config.id only used for the first time
+        let self_node_id = if is_open { sto.id } else { config.id };
 
-        if is_open {
-            // read id from sto, read listening addr from sto
-            builder = builder.node_id(sto.id);
-        } else {
-            // read id from config, read listening addr from config.
-            builder = builder.node_id(config.id).addr(
-                config
-                    .raft_api_addr()
-                    .await
-                    .map_err(|e| MetaError::Fatal(AnyError::new(&e)))?,
-            );
-        }
-
+        // use ip:port to start grpc listening
+        builder = builder
+            .node_id(self_node_id)
+            .addr(config.raft_api_listen_host_string());
```

Comment:
What about just passing in a tuple `(hostname, port)` into `addr()`? So that you do not need to parse the `host:port` string again in `start_grpc()`.